### DISCUSSION TITLE
When I use useFetch, I want this request to happen only once on the server side

### DISCUSSION THREAD

**Discussion by SnowGuest** [nuxt/nuxt#17437] [Questions] [answered]
As the title suggests, I hope to use useFetch or other alternative request methods to only make a request on the server side to render the string and return it to the browser, but when I use useFetch, I accidentally initiate a request in the browser. request, after deployment

```js
//  pages/search.vue
    const search = new search_Account(id);
    await search.onCreated();
```

```js
// pages/search.ts
export defalut class search_Account{
 public async onCreated(): Promise<void> {
        try {
          const result = await useFetch("api",{
           headers:{}
          })
         } catch(error){

         }
}
```

The request should only be made once on the server side, and the request should not be made again on the client side. In the development environment, the result is the same as the expected result. But after building and deploying to the server, The browser actually made the request again.

I don't know if useFetch has to be used in <script setup>, maybe it's a problem with my code, hope it gets answered🤔🤔

**Comment by Flanker72** [comment, marked as answer]
You may try to use `useAsyncData` instead of `useFetch`, for me in such a case it works perfectly.

**Reply by SnowGuest** [reply]
Thanks for your answer, your answer solved my problem
But I also found out where my problem is
